src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_Format.txt

```diff
@@ -639,132 +594,963 @@ Error({Kind:ErrorKind.InvalidArgument})
+Errors: Error 0-54: The function 'Text' has some invalid arguments.|Warning 16-41: Incorrect format specifier for 'Text'.
+
+﻿// Text function consistency with Excel
+//  
+// We do not support all the formats that Excel does.  For the moment, we are blocking several patterns that 
+// we may want to support in the future:
+//   1. ? - Whitespace padding digit (same as 0 with a space instead of a 0)
+//   2. #/ 0/ ?/ - Fractions (/ is too common to ban outright, only blocked if after a number placeholder)
+//   3. [] with anything other than [@-ll-RR] - Conditionals, currency symbols, output locale, elapsed time
+//   4. _ - Whitespace padding
+//   5. * - Alignment
+//   6. third ; - Fourth string based optional section
+//   7. "General" - General keyword (case insensitive)
+//   8. @ - Embeddable version of "General"
+//   9. : with anything other than time placeholders - Colon
+// All of these patterns can be embedded with escaping (backslash or double quotes)
 
->> Text(1234.5678, "[$-pt-BR]#.##0,00", "en-US")
+// Basic usage
+>> Text(1234567.1234567, "###")
+"1234567"
+
+>> Text(1234567.1234567, "000")
+"1234567"
+
+>> Text(1234567.1234567, "###.###")
+"1234567.123"
+
+>> Text(1234567.1234567, "#,###.###")
+"1,234,567.123"
+
+>> Text(1234567.1234567, "000,000,000")
+"001,234,567"
+
+>> Text(-1234567.1234567, "###")
+"-1234567"
+
+>> Text(-1234567.1234567, "000")
+"-1234567"
+
+>> Text(-1234567.1234567, "###.###")
+"-1234567.123"
+
+>> Text(-1234567.1234567, "#,###.###")
+"-1,234,567.123"
+
+>> Text(-1234567.1234567, "000,000,000")
+"-001,234,567"
+
+>> Text(0, "###")
+""
+
+>> Text(0, "000")
+"000"
+
+>> Text(0, "###.###")
+""
+
+>> Text(0, "#,###.###")
+""
+
+>> Text(0, "000,000,000")
+"000,000,000"
+
+>> Text(1234.567, "#,##0.00")
 "1,234.57"
 
->> Text(1234.5678, "[$-pt-BR]#.##0,00", "fr-FR")
-"1 234,57"
+>> Text(1, "00000")
+"00001"
 
->> Text(1234.5678, "[$-en-US]#,##0.00", "pt-BR")
-"1.234,57"
+>> Text(12, "00000")
+"00012"
 
->> Text(1234.5678,"[$-en-US]0 # # # !!!", "en-US")
-"1 2 3 5 !!!"
+>> Text(123, "00000")
+"00123"
 
-// The thousands separator in the fi-FI locale is the no-break space (U+00A0)
->> Text(12345.6789, $"[$-fi-FI]#{Char(160)}##0.00", "en-US")
-"12,345.68"
+>> Text(1234, "00000")
+"01234"
 
->> Text(12345.6789, $"[$-fi-FI]#{Char(160)}##0.00", "fi-FI")
-"12 345,68"
+>> Text(12345, "00000")
+"12345"
 
-// The thousands separator in the fr-FR locale is the narrow no-break space (U+202F)
->> Text(12345.6789, $"[$-fr-FR]#{ParseJSON("""\u202F""")}##0.00", "en-US")
-"12,345.68"
+>> Text(1234567.1234567, "0000000000")
+"0001234567"
 
->> Text(12345.6789, $"[$-fr-FR]#{ParseJSON("""\u202F""")}##0.00", "fr-FR")
-"12 345,68"
+>> Text(1234567.1234567, "##########")
+"1234567"
 
->> Text(1234.5678, "[$-en-US]", "en-US")
-"1234.5678"
+// Escape with backslash.
+>> Text(1234567.1234567, "\#,###.###")
+"#,1234567.123"
 
->> Text(1234.5678, "[$-fr-FR]", "fr-FR")
-"1234,5678"
+>> Text(1234567.1234567, "\\#,###.###")
```

Comment:
>Text(1234567.1234567, "\\#,###.###") [](http://example.com/codeflow?start=3&length=36)

Lots of those new tests are not supported in Power Apps today (which is what uses PowerFxV1CompatDisabled). Can we add them to a different file, so it will be easier to exclude them in that platform until the support is added?

Reply:
move to new test file.